WorkloadManagementSystem/Service/JobManagerHandler.py

```diff
@@ -240,6 +246,22 @@ def __deleteJob( self, jobID ):
     if not result['OK']:
       gLogger.warn( 'Failed to delete job from the TaskQueue' )
 
+    # if it was the last job for the pilot, clear PilotsLogging about it
+    result = gPilotAgentsDB.getPilotsForJobID( jobID )
+    if not result['OK']:
+      return result
+    for pilot in result['Value']:
+        res = gPilotAgentsDB.getJobsForPilot( pilot[ 'PilotID' ] )
+        if not res['OK']:
+            return res
+        if not res['Value']: # if list of jobs for pilot is empty, delete pilot and pilotslogging
+            ret = gPilotAgentsDB.deletePilot( pilot[ 'PilotID' ] )
```

Comment:
This is deleting pilots for which no jobs are registered. This should never happen in this context. But if it happens, it would be nice to see the pilot output for a possible reason.